vizro-core/src/vizro/_vizro.py

```diff
@@ -16,25 +16,35 @@
 class Vizro:
     """The main class of the `vizro` package."""
 
-    _user_assets_folder = Path.cwd() / "assets"
-    _lib_assets_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), "static")
-
-    def __init__(self):
+    def __init__(self, **kwargs):
         """Initializes Dash."""
-        _js, _css = _append_styles(self._lib_assets_folder, STATIC_URL_PREFIX)
-        self.dash = dash.Dash(
-            use_pages=True,
-            pages_folder="",
-            external_scripts=_js,
-            external_stylesheets=_css,
-            assets_folder=self._user_assets_folder,
+        self.dash = dash.Dash(**kwargs, use_pages=True, pages_folder="")
+
+        # Include Vizro assets (in the static folder) as external scripts and stylesheets. We extend self.dash.config
+        # objects so the user can specify additional external_scripts and externaL_stylesheets via kwargs.
+        vizro_assets_folder = Path(__file__).with_name("static")
+        static_url_path = self.dash.config.requests_pathname_prefix + STATIC_URL_PREFIX
+        vizro_css = self._get_external_assets(static_url_path, vizro_assets_folder, "css")
+        vizro_js = [
+            {"src": path, "type": "module"}
+            for path in self._get_external_assets(static_url_path, vizro_assets_folder, "js")
+        ]
+        self.dash.config.external_stylesheets.extend(vizro_css)
+        self.dash.config.external_scripts.extend(vizro_js)
+
+        # Serve all assets (including files other than css and js) that live in vizro_assets_folder at the
+        # route /vizro. Based on code in Dash.init_app that serves assets_folder. This respects the case that the
+        # dashboard is not hosted at the root of the server, e.g. http://www.example.com/dashboard/vizro.
+        blueprint_prefix = self.dash.config.routes_pathname_prefix.replace("/", "_").replace(".", "_")
+        self.dash.server.register_blueprint(
+            flask.Blueprint(
+                f"{blueprint_prefix}vizro_assets",
+                self.dash.config.name,
+                static_folder=vizro_assets_folder,
+                static_url_path=static_url_path,
```

Comment:
Just to understand, this now works, as it dynamically adjusts with the `self.dash.config.requests_pathname_prefix`, something that it did not do before right?

Reply:
Yes exactly!